config/v1/types_infrastructure.go

```diff
@@ -101,6 +101,12 @@ type InfrastructureStatus struct {
 	// +kubebuilder:default=HighlyAvailable
 	// +kubebuilder:validation:Enum=HighlyAvailable;SingleReplica
 	InfrastructureTopology TopologyMode `json:"infrastructureTopology"`
+
+	// cpuPartitioning expresses that CPU partitioning is a currently enabled feature in the cluster.
+	// CPU Partitioning means that this cluster can support partitioning workloads to specific CPU Sets
+	// +kubebuilder:default=None
```

Comment:
I'm betting we'll need some kind of admission plugin to detect the cluster state and handle upgraded clusters properly.  After the first release this lands in, we could simplify, but that first upgrade is going to have to adjust the state.

Reply:
Yeah, since it's only a one time event and only for single node might be to just make those parts of the cluster aware for the upgrade cycle and remove it after 4.12.

What do you think of this approach?

The thinking is NTO does the update of status under the single node upgrade event

https://github.com/openshift/cluster-node-tuning-operator/pull/431/files#diff-b1da0f702ee090192a7506701ce502f780265a647ee3a7b29ac0ce37d2fea1aaR259-R302

and the admission controller is aware of single node when doing it's logic checks for pod modifications

https://github.com/openshift/kubernetes/pull/1312/files#diff-2aedbe2d2f5c230833b2587136f8dbdc942db62589d593f8a781a6968054ef6bR279-R292

Reply:
Hey @deads2k , what do you think of the approach above, would that be in-line with a solution or do you think we'll need an admission plugin?

Reply:
> Yeah, since it's only a one time event and only for single node might be to just make those parts of the cluster aware for the upgrade cycle and remove it after 4.12.

> What do you think of this approach?

I think that if the node tuning operator is off for a few hours during upgrade, the default set here will return None on a cluster that is actually AllNodes.  It seems like that would cause a problem.